src/ws/cockpitchannelresponse.c

```diff
@@ -82,49 +82,47 @@ cockpit_channel_inject_perform (CockpitChannelInject *inject,
                                 CockpitTransport *transport)
 {
   static const gchar *marker = "<head>";
-  CockpitWebFilter *filter;
   CockpitCreds *creds;
-  gchar *prefixed_application = NULL;
-  const gchar *checksum;
-  GString *str;
-  GBytes *base;
+  g_autofree gchar *prefixed_application = NULL;
 
-  if (!inject->base_path)
+  const gchar *url_root = cockpit_web_response_get_url_root (response);
+
+  if (!url_root && !inject->base_path)
     return;
 
-  str = g_string_new ("");
+  GString *str = g_string_new ("");
   creds = cockpit_web_service_get_creds (inject->service);
```

Comment:
Move this into the else branch, please, and bring the variable declaration with it.

Reply:
It's also used in the `then` branch, so I didn't move the initialization. But I did move the declaration.